apps/transformers/lib/validations/not_blank.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Transformers.Validations.NotBlank do
+  alias Transformers.Validations.ValidationStatus
+
+  def check(status, parameters, field) do
+    if ValidationStatus.has_error?(status, field) do
+      status
+    else
+      check_if_blank(status, parameters, field)
+    end
+  end
+
+  defp check_if_blank(status, parameters, field) do
+    value = Map.get(parameters, field)
+
+    case value do
+      nil -> ValidationStatus.add_error(status, field, "Missing or empty field")
+      value when is_binary(value) -> check_if_blank_string(status, field, value)
+      value when is_list(value) -> check_if_blank_list(status, field, value)
+      _ -> add_unsupported_type_error(status, field)
+    end
```

Comment:
Project style usually calls for a function hydra here. Is there a specific reason for a case instead?

Reply:
Logically it's the same but I find the case statement more readable. It's easier for me to see that there's branching logic and go to the next step. With a hydra I have to pick out the hydra functions from the rest of the functions in the file, and it's harder to see what the cases are because the guards are in the middle of the function declaration.

Readability is not a one size fits all approach though, so other people may feel differently.

Reply:
```elixir
defp check_if_blank(status, parameters, field) do
  parameters
  |> Map.get(field)
  |> apply_validation()
end  

defp apply_validation(status, field, value) when is_nil(value), do: ValidationStatus.add_error(status, field, "Missing or empty field")

defp apply_validation(status, field, value) when is_binary(value), do: check_if_blank_string(status, field, value)

defp apply_validation(status, field, value) when is_list(value), do: check_if_blank_list(status, field, value)

defp apply_validation(_), do: add_unsupported_type_error(status, field)
```

I think I agree with you here, with the number of parameters that would need to be passed there are some benefits to being in the same scope too.